app/controllers/posts_controller.rb

```diff
@@ -6,5 +6,44 @@ def index
   def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
+    @comment = Comment.new
+    @like = Like.new
+  end
+
+  def create_comment
+    @comment = Comment.new(comment_params)
+    @comment.post = Post.find(params[:id])
+    @comment.author = current_user
+    redirect_to post_path if @comment.save
+  end
+
+  def create_like
+    @like = Like.new(author: current_user, post: Post.find(params[:id]))
+    redirect_to post_path if @like.save
+  end
```

Comment:
- [ ] Your app still uses these methods to create likes and comments. Kindly remove these methods and refactor your implementation to use the likes and comments controller you created.